python/lsst/jointcal/testUtils.py

```diff
@@ -0,0 +1,290 @@
+                skyWcs=None, fluxErrFraction=0.05, refCat=False):
+    """Return a catalog populated with fake, but reasonable, sources.
+
+    Centroids are placed on a uniform grid, errors are normally distributed.
+
+    Parameters
+    ----------
+    schema : `lsst.afw.table.Schema`
+        Pre-built schema to make the catalog from.
+    num : `int`
+        Number of sources to put in the catalog.
+    bbox : `lsst.afw.geom.Box2I`
+        Bounding box of the ccd to put sources in.
+    centroidKey : `lsst.afw.table.Key`
+        Key for the centroid field to populate.
+    xErrKey : `lsst.afw.table.Key`
+        Key for the xErr field to populate.
+    yErrKey : `lsst.afw.table.Key`
+        Key for the yErr field to populate.
+    shapeKey : `lsst.afw.table.Key`
+        Key for the shape field to populate.
+    instFluxKeyName : `str`
+        Name of instFlux field to populate (i.e. instFluxKeyName+'_flux')
+    skyWcs : `lsst.afw.geom.SkyWcs` or None, optional
+        If supplied, use this to fill in coordinates from centroids.
+    fluxErrFraction : `float`, optional
+        Fraction of instFlux to use for the instFluxErr.
+    refCat : `bool`, optional
+        Return a ``SimpleCatalog`` so that it behaves like a reference catalog?
+
+    Returns
+    -------
+    catalog : `lsst.afw.table.SourceCatalog`
+        The filled catalog.
+    """
+    table = lsst.afw.table.SourceTable.make(schema)
+    table.defineCentroid('centroid')
+    table.defineShape('shape')
+    table.defineInstFlux(instFluxKeyName)
+    if refCat:
+        catalog = lsst.afw.table.SimpleCatalog(table)
+    else:
+        catalog = lsst.afw.table.SourceCatalog(table)
+
+    instFlux = np.random.random(num)
+    instFluxErr = instFlux * fluxErrFraction
+    xx = np.linspace(bbox.getMinX(), bbox.getMaxX(), int(np.sqrt(num)))
+    yy = np.linspace(bbox.getMinY(), bbox.getMaxY(), int(np.sqrt(num)))
+    xv, yv = np.meshgrid(xx, yy)
+    vx = np.random.normal(scale=0.1, size=num)
+    vy = np.random.normal(scale=0.1, size=num)
+
+    # make all the sources perfectly spherical, for simplicity.
+    mxx = 1
+    myy = 1
+    mxy = 0
+
+    for i, (x, y) in enumerate(zip(xv.ravel(), yv.ravel())):
+        record = catalog.addNew()
+        record.set('id', i)
+        record.set(centroidKey, lsst.afw.geom.Point2D(x, y))
+        record.set(shapeKey, lsst.afw.geom.ellipses.Quadrupole(mxx, myy, mxy))
+
+    if skyWcs is not None:
+        lsst.afw.table.updateSourceCoords(skyWcs, catalog)
+
+    catalog[xErrKey] = vx
+    catalog[yErrKey] = vy
+    catalog[instFluxKeyName + '_flux'] = instFlux
+    catalog[instFluxKeyName + '_fluxSigma'] = instFluxErr
+
+    return catalog
+
+
+def getMeasuredStarsFromCatalog(catalog, pixToFocal):
+    """Return a list of measuredStars built from a catalog.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SourceCatalog`
+        The table to get sources from.
+    pixToFocal : `lsst.afw.geom.Transform`
```

Comment:
I think you mean `lsst.afw.geom.TransformPoint2ToPoint2`

`Transform` is not actually a class in Python (though it is a templated class in C++)